lib/trino-parquet/src/main/java/io/trino/parquet/reader/FilteredRowRanges.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import org.apache.parquet.internal.filter2.columnindex.RowRanges;
+
+import static java.util.Objects.requireNonNull;
+
+public record FilteredRowRanges(RowRanges parquetRowRanges, long rowCount)
```

Comment:
This record class is unnecessary. It just stores `parquetRowRanges.rowCount()` in a separate field, which callers can access directly from the underlying `parquetRowRanges`.

Reply:
Note that `rowCount()` is not a simple getter, it performs some computations before giving back the result. Parquet's `RowRanges` is a mutable class whereas for our usage we wanted to access `rowCount` as if it is immutable and free to read.
This class also gains further functionality in subsequent commit for flat column reader, introducing it here helps to make that commit a bit smaller.